GroceryList/app/src/main/java/com/example/siyer/grocerylist/MainActivity.java

```diff
@@ -0,0 +1,113 @@
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.EditText;
+import android.widget.ListView;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import java.util.ArrayList;
+
+public class MainActivity extends AppCompatActivity {
+
+    ArrayList<String> toDoList = new ArrayList<>();//List with all the items
+    ArrayAdapter<String> itemsAdapter;//Array Adapter for the list
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        View addListnr = findViewById(R.id.addbtn);//Listener for the add button
+        final EditText addedGrocery = (EditText) findViewById(R.id.groceryItem);
+        addListnr.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {//Adding the text to the list
+                toDoList.add(0,addedGrocery.getText().toString());
+                itemsAdapter.notifyDataSetChanged();
+                addedGrocery.setText("");//Clear the text field
+            }
+        });
+        final AlertDialog.Builder alertDialogBuilder = new AlertDialog.Builder(this);//Alert Dialog with Edit and Delete options
+        final AlertDialog.Builder alertEdit = new AlertDialog.Builder(this);//Alert Dialog with Edit text field
+        itemsAdapter = new ArrayAdapter<String>(this, android.R.layout.simple_list_item_1, toDoList);
+        ListView listView = (ListView) findViewById(R.id.groceryList);
+        listView.setAdapter(itemsAdapter);
+        listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+            @Override
+            public void onItemClick(AdapterView<?> parent, View view, final int position, long id) {
+                final String selectedItem = parent.getItemAtPosition(position).toString();
+                //Toast.makeText(getApplicationContext(), ((TextView) view).getText(), Toast.LENGTH_SHORT).show();
+
+                alertDialogBuilder.setMessage("Do you want to edit or delete the item?");
+
+                alertDialogBuilder.setPositiveButton("Edit", new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface arg0, int arg1) {
+                        //Toast.makeText(MainActivity.this, "You clicked edit button", Toast.LENGTH_LONG).show();
+
+                        alertEdit.setMessage("Enter the item");
+                        final EditText editedItem = new EditText(MainActivity.this);
+
+                        editedItem.setInputType(InputType.TYPE_CLASS_TEXT);
+                        editedItem.setText(selectedItem);
+                        alertEdit.setView(editedItem);
+
+                        alertEdit.setPositiveButton("Ok", new DialogInterface.OnClickListener() {
```

Comment:
you would have to use getString(R.string.ok)

R.id.ok is actually an integer that getStrings knows how to process to return a String

Alt + Enter -> Extract string resource should add the code for you by default.